### DISCUSSION TITLE
How can pie charts implement such labels as follows?

### DISCUSSION THREAD

**Discussion by saint3347** [airbnb/visx#1087] [Q&A]
![image](https://user-images.githubusercontent.com/7347133/102736891-f4fc9180-4380-11eb-9cdb-fef0c6376070.png)

**Comment by saint3347** [comment]
This variation of a donut chart demonstrates how to add labels with lines.

**Comment by williaster** [comment]
Hey @saint3347 thanks for checking out `visx`! Sorry for the delayed response (holidays 🎁 )

I've not seen this approach for labels implemented yet with `visx` (if anyone has examples I'd love to see them!). You get part of the way there with `@visx/annotation`s, see this modified https://airbnb.io/visx/pies [codesandbox](https://codesandbox.io/s/clever-wildflower-dz4n0?file=/Example.tsx). One note is that the `Connector` component currently creates a 45° angle from the reference point, whereas your example has lines which are perpendicular to the tangent of the acs. We'd need to update `Connector` to allow override of the middle point in order to match exactly.

<img src="https://user-images.githubusercontent.com/4496521/103809314-2bd3d800-500e-11eb-814f-d86ccf7640ff.png" width="500" />

The trickiest part is the positioning of the labels so they don't overlap, your example is using some algorithm to do that, whereas my example doesn't render a label if the arc size is too small. If you end up implementing it, consider contributing it back to `visx` as I'm sure others would love to leverage it and I would happily review a PR!

**Comment by techniq** [comment]
A few d3 examples that look to be implemented similarly to @saint3347's example:
- https://observablehq.com/@stefanreifenberg/donut-chart-from-the-d3-js-graph-gallery
- https://observablehq.com/@mingzhaolin/build-pie-chart-ct-books
- http://bl.ocks.org/dbuezas/9572040
